diff --git a/src/pins/tang/meson.build b/src/pins/tang/meson.build
index 774d0fd..fdabf4a 100644
--- a/src/pins/tang/meson.build
+++ b/src/pins/tang/meson.build
@@ -4,16 +4,19 @@ actv = find_program(
   required: false
 )
 kgen = find_program(
+  '/usr/lib/x86_64-linux-gnu/tangd-keygen',
   join_paths(libexecdir, 'tangd-keygen'),
   '/usr/libexec/tangd-keygen',
   required: false
 )
 updt = find_program(
+  '/usr/lib/x86_64-linux-gnu/tangd-update',
   join_paths(libexecdir, 'tangd-update'),
   '/usr/libexec/tangd-update',
   required: false
 )
 tang = find_program(
+  '/usr/lib/x86_64-linux-gnu/tangd',
   join_paths(libexecdir, 'tangd'),
   '/usr/libexec/tangd',
   required: false
@@ -31,6 +34,7 @@ if curl.found()
     env.append('PATH',
       join_paths(meson.source_root(), 'src'),
       meson.current_source_dir(),
+      '/usr/lib/x86_64-linux-gnu',
       '/usr/libexec',
       libexecdir,
       separator: ':'
